core/modules/filter/filter.module

```diff
@@ -2714,6 +2724,194 @@ function _filter_image_align($text) {
+              // Replace src with styled URL
+              $new_image = str_replace($src, $styled_url, $image);
+              $text = str_replace($image, $new_image, $text);
+            }
+          }
+        }
+      }
+    }
+  }
+  return $text;
+}
+
+/**
+ * Helper function to determine image URI from src.
+ */
+function _filter_image_style_uri($src) {
+  global $base_path;
+
+  $visible_wrappers = array_intersect_key(
+    file_get_stream_wrappers(STREAM_WRAPPERS_WRITE),
+    file_get_stream_wrappers(STREAM_WRAPPERS_VISIBLE)
+  );
+
+  $src = parse_url($src, PHP_URL_PATH);
+  $src = trim($src, '/');
+
+  foreach ($visible_wrappers as $scheme => $data) {
+    $class = file_stream_wrapper_get_class($scheme);
+    $wrapper = new $class();
+    if (method_exists($wrapper, 'getDirectoryPath')) {
+      $base_dir = trim($base_path . $wrapper->getDirectoryPath(), '/');
+      if (!empty($base_dir) && strpos($src, $base_dir) === 0) {
+        return $scheme . '://' . substr($src, strlen($base_dir) + 1);
+      }
+    }
+  }
+  return FALSE;
+}
+
+
+/**
+ * Filter tips callback for image style filter.
+ */
+function _filter_image_style_tips($filter, $format, $long = FALSE) {
+  if ($long) {
+    $enabled_styles = array_filter($filter->settings['image_styles']);
+    $image_styles = image_styles();
+    $style_names = array();
+
+    foreach ($enabled_styles as $style_name) {
+      if (isset($image_styles[$style_name])) {
+        $style_names[] = $image_styles[$style_name]['label'];
+      }
+    }
+
+    if (!empty($style_names)) {
+      return t('Images can be displayed using the following styles: %styles.', array(
+        '%styles' => implode(', ', $style_names),
+      ));
+    }
+    return t('No image styles have been enabled for this text format.');
+  }
+
+  return t('Images can be displayed using configured image styles.');
+}
+
+/**
+ * Implements hook_form_BASE_FORM_ID_alter() for node edit forms.
+ */
+function filter_form_node_form_alter(&$form, &$form_state, $form_id) {
+  filter_form_alter_common($form, $form_state, $form_id);
+}
+
+/**
+ * Implements hook_form_BASE_FORM_ID_alter() for block add forms on a layout
+ * screen.
+ */
+function filter_form_block_add_block_form_alter(&$form, &$form_state, $form_id) {
+  filter_form_alter_common($form, $form_state, $form_id);
+}
+
+/**
+ * Implements hook_form_BASE_FORM_ID_alter() for block configuration forms.
+ */
+function filter_form_block_admin_configure_alter(&$form, &$form_state, $form_id) {
+  filter_form_alter_common($form, $form_state, $form_id);
+}
+
+/**
+ * Implements hook_form_BASE_FORM_ID_alter() for block configuration forms.
+ */
+function filter_form_layout_block_configure_form_alter(&$form, &$form_state, $form_id) {
+  filter_form_alter_common($form, $form_state, $form_id);
+}
+
+/**
+ * Common form alter logic for both node and block add forms.
+ */
+function filter_form_alter_common(&$form, &$form_state, $form_id) {
```

Comment:
This could be avoided by using `hook_element_info_alter()` and adding to the `text_format` element like this (if this were moved to image.module):

```
function image_element_info_alter($elements) {
  $elements['text_format']['#attached']['js'][] = array($settings, 'setting');
```